Spigot-Server-Patches/0230-Max-exp-value-when-merging-orbs.patch

```diff
@@ -0,0 +1,67 @@
+From 9088a4ae9e1826b6479c1c091e51ee2dc41495e0 Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Mon, 31 Jul 2017 01:35:54 -0500
+Subject: [PATCH] Max exp value when merging orbs
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index 54d081fd..e0977149 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -428,4 +428,10 @@ public class PaperWorldConfig {
+         disableCreeperLingeringEffect = getBoolean("disable-creeper-lingering-effect", false);
+         log("Creeper lingering effect: " + disableCreeperLingeringEffect);
+     }
++
++    public int expMergeMaxValue;
++    private void expMergeMaxValue() {
++        expMergeMaxValue = getInt("exp-merge-max-value", 0);
++        log("Experience Merge Max Value: " + expMergeMaxValue);
++    }
+ }
+diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
+index 9dfc2b4d..e9aeb5f2 100644
+--- a/src/main/java/net/minecraft/server/World.java
++++ b/src/main/java/net/minecraft/server/World.java
+@@ -1,4 +1,4 @@
+-	package net.minecraft.server;
++package net.minecraft.server;
+ 
+ import com.destroystokyo.paper.event.server.ServerExceptionEvent;
+ import com.destroystokyo.paper.exception.ServerInternalException;
+@@ -1146,16 +1146,26 @@ public abstract class World implements IBlockAccess {
+             EntityExperienceOrb xp = (EntityExperienceOrb) entity;
+             double radius = spigotConfig.expMerge;
+             if (radius > 0) {
++                // Paper start
++                int maxValue = paperConfig.expMergeMaxValue;
+                 List<Entity> entities = this.getEntities(entity, entity.getBoundingBox().grow(radius, radius, radius));
+                 for (Entity e : entities) {
+-                    if (e instanceof EntityExperienceOrb) {
+-                        EntityExperienceOrb loopItem = (EntityExperienceOrb) e;
+-                        if (!loopItem.dead) {
+-                            xp.value += loopItem.value;
+-                            loopItem.die();
+-                        }
++                    if (!(e instanceof EntityExperienceOrb)) {
++                        continue;
++                    }
++                    EntityExperienceOrb loopItem = (EntityExperienceOrb) e;
++                    if (loopItem.dead || (maxValue > 0 && loopItem.value >= maxValue)) {
++                        continue;
++                    }
++                    xp.value += loopItem.value;
++                    if (xp.value > maxValue) {
```

Comment:
I think you are misunderstanding what I am stating. See the full context what this boolean check is doing.
We are not checking whether xp.value is bigger than 0; this would check that maxValue is greater than zero.
(This change is necessary as a result of your last commit).